src/global/useConfig.tsx

```diff
@@ -0,0 +1,18 @@
+import { createContext, useContext, useState } from 'react';
+import { useQuery } from 'react-query';
+
+const ConfigContext = createContext({});
+
+export const ConfigProvider = ({ children }: { children: any }) => {
+	// only need to do this once to get values from runtime api route
+	const [config, setConfig] = useState({});
+	useQuery('config', async () => {
+		const res = await fetch('http://localhost:3000/api');
+		const data = await res.json();
+		console.log('use query', data);
+		setConfig(data);
+	});
```

Comment:
Will this need a dependency array?

Reply:
This is using useQuery from react-query so it works a little different to useEffect
it also caches by default

that been said we do have to tweak some work in `layout.tsx` having multiple re-renders that aren't needed.
This is an existing issue already.

Reply:
I hear what you're saying - ensure as much as possible it gets called once. once is all that's needed